openstack/datadog_checks/openstack/openstack.py

```diff
@@ -945,32 +928,58 @@ def get_all_servers(self, i_key, collect_all_tenants, filter_by_host=None):
                 except KeyError as e:
                     self.log.debug("Server: %s has already been removed from the cache", new_server['server_id'])
 
-        return self.server_details_by_id
+    def filter_excluded_servers(self):
+        proj_list = set([])
+        if self.exclude_server_id_rules:
+            # Filter out excluded servers
+            for exclude_id_rule in self.exclude_server_id_rules:
+                for server_id in self.server_details_by_id.keys():
+                    if re.match(exclude_id_rule, server_id):
+                        del self.server_details_by_id[server_id]
+
+        for server in self.server_details_by_id.iteritems():
+            proj_list.add(server[1].get('project_name'))
```

Comment:
why `1` ?

Reply:
Its because we are using `iteritems` here, so the first `0` element is just the ID of the server, and `1` is the actual details about that server.

Reply:
let's unpack: `for _, server in self.server_details_by_id.iteritems():`